sale_product_price/models/sale_order_line.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    pricelist_id = fields.Many2one('product.pricelist', 'Pricelist')
+
+    @api.onchange('product_template_id')
+    def _onchange_product_id_set_pricelist(self):
+        """
+        Automatically selects the pricelist with the lowest price for
+        the selected product.
+        """
+        if not self.product_template_id or not self.order_id:
+            return
+
+        pricelists = self.env['product.pricelist'].with_context({
```

Comment:
Please explain in brief use of with context and search in this

Reply:
for finding lowest price pricelist for selected product, we are passing context means we are passing customer name and date from sales order to filter record from pricelist after that search method will search(which will take less time then without context search method) records base on domain.